gitpod-io/gitpod
# [JetBrains] Git-ignore the proto-generated Java files from Supervisor API

## Description

Git-ignore the proto-generated Java files from Supervisor API.

The decision to commit those files was based on the fact that Go and TypeScript generated files were also versioned ([More details in this comment](https://github.com/gitpod-io/gitpod/pull/5687#issuecomment-935893306)). Still, we've noticed there's no actual need for them to be committed, so we can generate them on demand while leaving them git-ignored.

- Note: This PR has two commits, and it should be merged to the main branch like this because the second one is a fix of https://github.com/gitpod-io/gitpod/pull/15273 ([More details in this comment](https://github.com/gitpod-io/gitpod/pull/15468#issuecomment-1362574345))

## How to test

1. Confirm if Werft Build passes.
2. Start a new workspace from this PR, using IntelliJ IDEA, then confirm if smart-completion is working from JetBrains Backend plugin codebase.
3. Open a new workspace from https://github.com/Gitpod-Samples/spring-petclinic, on the Preview Environment of this PR, using IntelliJ IDEA, and confirm if the backend-plugin has been loaded (it's enough to check if our customizations on Control Center are loaded).

## Release Notes

```release-note
NONE
```

## Werft options:

      If enabled this will build `install/preview`
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
